python/cuspatial/cuspatial/core/gis.py

```diff
@@ -74,26 +78,92 @@ def directed_hausdorff_distance(xs, ys, points_per_space):
     >>> result = cuspatial.directed_hausdorff_distance(
             [0, 1, 0, 0], # xs
             [0, 0, 1, 2], # ys
-            [2,    2],    # points_per_space
+            [0,    2],    # offsets
         )
     >>> print(result)
              0         1
         0  0.0  1.414214
         1  2.0  0.000000
     """
 
-    num_spaces = len(points_per_space)
+    num_spaces = len(offsets)
     if num_spaces == 0:
         return DataFrame()
     xs, ys = normalize_point_columns(as_column(xs), as_column(ys))
     result = cpp_directed_hausdorff_distance(
-        xs, ys, as_column(points_per_space, dtype="int32"),
+        xs, ys, as_column(offsets, dtype="int32"),
     )
     result = result.data_array_view
     result = result.reshape(num_spaces, num_spaces)
     return DataFrame.from_gpu_matrix(result)
 
 
+def directed_polygon_distance(xs, ys, offsets):
+    """Compute the directed polygon-point distances between all pairs of
+    polygons.
+
+    Parameters
+    ----------
+    xs
+        column of x-coordinates
+    ys
+        column of y-coordinates
+    offsets
+        number of points in each space
+
+    Returns
+    -------
+    result : cudf.DataFrame
+        The pairwise directed distance matrix with one row and one column per
+        input polygon; the value at row `i`, column `j` represents the
+        polygon-point distance from polygon i to any vertex in polygon j.
```

Comment:
?

```suggestion
        minimum polygon-point distance from polygon i to any vertex in polygon j.
```

Reply:
From which point in polygon i is the distance measured? Is it really "the minimum distance from any vertex in i to any vertex in j"?

If so, why would the distance from i to j be different from the distance from j to i?

Reply:
If the question is "why would..." the answer is: because it was more computationally efficient and required less redundant reads.
If the question had been "why _should_..." the answer is not so simple.

That said, this is the pattern that Hausdorff follows right now, and it's because its more efficient _on the computation side_, but might be less efficient on the consumers end, depending on how it's used. we should probably approach all/most distance metric APIs similarly, and Hausdorff set the precedence.

This requires further discussion.

Reply:
No, the question was another way of asking "why is this a directed distance?". And the reason is because it's the distance from vertices of A to edges of B, not from edges of A to edges of B. If it were the latter, it would be a symmetric distance. This is just asking for clarity in the docs of the geometric reason for having two different distances per pair.